### PR TITLE
KAFKA-10679: Migrate upgrade changes from site to kafka/docs

### PR DESCRIPTION
**Pull request by bbejeck**
During the AK website upgrade, changes made to `kafka-site` weren't migrated back to `kafka-docs.`

This PR is an initial attempt at porting the changes to `kafka/docs,` but it does not include the streams changes. Those will come in a separate PR.

For the most part, the bulk of the changes in the PR are cosmetic.  Only the `introduction.html` has substantial changes, but it's a direct port from the live documentation.

For testing:

1. I reviewed the PR diffs 
2. Rendered the changes locally

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/design.html [hunk 1]
```diff
@@ -508,7 +506,7 @@ <h4><a id="design_compactionbasics" href="#design_compactionbasics">Log Compacti
     the log, even if the message with that offset has been compacted away; in this case this position is indistinguishable from the next highest offset that does appear in the log. For example, in the picture above the
     offsets 36, 37, and 38 are all equivalent positions and a read beginning at any of these offsets would return a message set beginning with 38.
     <p>
-    Compaction also allows for deletes. A message with a key and a null payload will be treated as a delete from the log. Such a record is sometimes referred to as a <i>tombstone</i>. This delete marker will cause any prior message with that key to be removed (as would any new
+    Compaction also allows for deletes. A message with a key and a null payload will be treated as a delete from the log. This delete marker will cause any prior message with that key to be removed (as would any new
```
Comment (mjsax) [inline docs/design.html]: Why was this sentence removed?

File: docs/design.html [hunk 2]
```diff
@@ -543,25 +544,25 @@ <h4><a id="design_compactiondetails" href="#design_compactiondetails">Log Compac
     (assuming 1k messages).
     </ol>
     <p>
-    <h4><a id="design_compactionconfig" href="#design_compactionconfig">Configuring The Log Cleaner</a></h4>
+    <h4 class="anchor-heading"><a id="design_compactionconfig" class="anchor-link"></a><a href="#design_compactionconfig">Configuring The Log Cleaner</a></h4>
 
     The log cleaner is enabled by default. This will start the pool of cleaner threads.
     To enable log cleaning on a particular topic, add the log-specific property
-    <pre class="brush: text;"> log.cleanup.policy=compact</pre>
+    <pre class="language-text"> log.cleanup.policy=compact</code></pre>
```
Comment (mjsax) [inline docs/design.html]: Seems we are missing a opening `<code>` tag? Similar the next two changes below.

File: docs/introduction.html [hunk 3]
```diff
@@ -18,198 +18,203 @@
...
-
-  <h4><a id="intro_topics" href="#intro_topics">Topics and Logs</a></h4>
-  <p>Let's first dive into the core abstraction Kafka provides for a stream of records&mdash;the topic.</p>
-  <p>A topic is a category or feed name to which records are published. Topics in Kafka are always multi-subscriber; that is, a topic can have zero, one, or many consumers that subscribe to the data written to it.</p>
-  <p> For each topic, the Kafka cluster maintains a partitioned log that looks like this: </p>
-  <img class="centered" src="/{{version}}/images/log_anatomy.png">
-
-  <p> Each partition is an ordered, immutable sequence of records that is continually appended to&mdash;a structured commit log. The records in the partitions are each assigned a sequential id number called the <i>offset</i> that uniquely identifies each record within the partition.
-  </p>
+  <h4 class="anchor-heading">
```
Comment (mjsax) [inline docs/introduction.html:21]: Cannot really verify those changes. But I guess, should be ok.

File: docs/ops.html [hunk 4]
```diff
@@ -78,37 +68,31 @@ <h4><a id="basic_ops_restarting" href="#basic_ops_restarting">Graceful shutdown<
...
-  </pre>
+  <pre class="line-numbers"><code class="language-bash">  &gt; bin/kafka-preferred-replica-election.sh --bootstrap-server broker_host:port</code></pre>
 
-  <h4><a id="basic_ops_racks" href="#basic_ops_racks">Balancing Replicas Across Racks</a></h4>
+  <h4 class="anchor-heading"><a id="basic_ops_racks" class="anchor-link"></a><a href="#basic_ops_racks">Balancing Replicas Across Racks</a></h4>
   The rack awareness feature spreads replicas of the same partition across different racks. This extends the guarantees Kafka provides for broker-failure to cover rack-failure, limiting the risk of data loss should all the brokers on a rack fail at once. The feature can also be applied to other broker groupings such as availability zones in EC2.
   <p></p>
   You can specify that a broker belongs to a particular rack by adding a property to the broker config:
-  <pre class="brush: text;">   broker.rack=my-rack-id</pre>
+  <pre class="language-text">   broker.rack=my-rack-id</code></pre>
```
Comment (mjsax) [inline docs/ops.html]: Missing opening `<code>` tag

File: docs/ops.html [hunk 5]
```diff
@@ -1658,9 +1564,9 @@ <h4><a id="kafka_streams_monitoring" href="#kafka_streams_monitoring">Streams Mo
   Use the following configuration option to specify which metrics
   you want collected:
 
-<pre>metrics.recording.level="info"</pre>
+<pre>metrics.recording.level="info"</code></pre>
```
Comment (mjsax) [inline docs/ops.html]: Missing opening `<code>` tag

File: docs/quickstart-docker.html [hunk 6]
```diff
@@ -0,0 +1,204 @@
+<!--
```
Comment (mjsax) [inline docs/quickstart-docker.html:1]: Cannot really verify this one.

File: docs/quickstart-zookeeper.html [hunk 7]
```diff
@@ -0,0 +1,277 @@
+<!--
```
Comment (mjsax) [inline docs/quickstart-zookeeper.html:1]: Cannot really verify this one.

**Review by mjsax — APPROVED** [review]
Feel free to merge.

For the comments: I would assume those are issue in other branches, too. Thus, might be best to do small follow up PR to address them, so we can cherry-pick the fix to other branches, too.

File: docs/design.html (hunk 1 above)
Comment (bbejeck) [reply docs/design.html]: I don't know, I missed this change when reviewing, I'll move it back

File: docs/design.html (hunk 2 above)
Comment (bbejeck) [reply docs/design.html]: ack

**Comment by bbejeck** [comment]
merged #9551 into trunk

**Comment by bbejeck** [comment]
cherry-picked to 2.7

**Pull request merged**